### PR TITLE
Refactor WindowNode.partition/order/frame into WindowNode.Specification

### PR DESCRIPTION
**Pull request by ebd2**
This is a precursor to continuing window function work.

### CODE REVIEW

**Comment by ebd2** [comment]
This passes the un-merged product tests in #5349.

**Comment by electrum** [comment]
This seems fine, but I don't see any reason to remove all the accessor methods in `WindowNode`.

Also, any reason not to squash the "Replace redundant construction" commit into the "Refactor WindowNode constructor" commit?

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java [hunk 1]
```diff
@@ -173,11 +168,12 @@ public int getPreSortedOrderPrefix()
         private final Map<Symbol, SortOrder> orderings;
         private final Frame frame;
 
+        @JsonCreator
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java]: This can go in the first commit even though it's not used there

**Comment by electrum** [comment]
Merged, thanks!

**Comment by electrum** [comment]
I did not merge the removal of the accessor methods, as I'm not convinced it provides benefit. We can do so later using the "inline method" refactoring, if needed.

**Pull request closed without merging**